src/App.tsx

```diff
@@ -1,223 +1,188 @@
+import { Header } from './components/Header/Header';
+import { TodoInfo } from './components/TodoInfo/TodoInfo';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorTypes, FilterTypes } from './types/Enums';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterType, setFilterType] = useState('All');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodosIds, setDeletingTodosIds] = useState<number[]>([]);
+
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const handleButtonClickAll = () => {
+    setFilterType(FilterTypes.All);
+  };
+
+  const handleButtonClickActive = () => {
+    setFilterType(FilterTypes.ACTIVE);
+  };
+
+  const handleButtonClickCompleted = () => {
+    setFilterType(FilterTypes.COMPLETED);
+  };
+
+  const onSubmit = useCallback((event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle.trim()) {
+      setErrorMessage(ErrorTypes.EmptyTitle);
+      setTodoTitle('');
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    if (user) {
+      setTempTodo({
+        id: 0,
+        title: todoTitle,
+        completed: false,
+        userId: user.id,
+      });
+
+      addTodo(todoTitle, user.id)
+        .then(response => {
+          setTodos(prev => [...prev, {
+            id: response.id,
+            title: response.title,
+            completed: response.completed,
+            userId: response.userId,
+          }]);
+        })
+        .catch(() => setErrorMessage(ErrorTypes.UnableToAdd))
+        .finally(() => {
+          setIsLoading(false);
+          setTempTodo(null);
+          setTodoTitle('');
+        });
+    }
+  }, [todoTitle, user]);
+
+  const onDeleteTodo = useCallback(async (todoId: number) => {
+    try {
+      setDeletingTodosIds(prev => [...prev, todoId]);
+
+      await deleteTodo(todoId);
+
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    } catch (error) {
+      setErrorMessage('todo is not delete');
+    } finally {
+      setDeletingTodosIds(prev => prev.filter(id => id !== todoId));
+    }
+  }, []);
+
+  const clearCompleated = () => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        onDeleteTodo(todo.id);
+      }
+    });
+  };
+
+  const visibleTodos = useMemo(() => {
+    switch (filterType) {
+      case FilterTypes.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+
+      case FilterTypes.COMPLETED:
+        return todos.filter(todo => todo.completed);
+
+      case FilterTypes.All:
+        return todos;
+
+      default:
+        return todos;
+    }
```

Comment:
In the future, better to move a big parts of code into separate files, it's extremely impro code readability